modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java

```diff
@@ -176,14 +176,14 @@ protected HttpRequest getHttpRequest(NHttpClientConnection conn, String path) {
                 request = new BasicHttpRequest(method, path, version != null ? version : HttpVersion.HTTP_1_1);
```

Comment:
Hi Kim,
When I going through including your previous commit also I see that you have created BasicHttpEntityEnclosingRequest [1] and then again create BasicHttpRequest if the request has no entity. This is unnecessary and, cause extra instructions which does not add value and hit slightly performance. Shall we move creation BasicHttpEntityEnclosingRequest to the place that makes the decision

[1] https://github.com/wso2/wso2-synapse/pull/1127/files#diff-33193add0edf667ac97380fd68849275R153